fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/BaseCompositeAvailabilitySpec.groovy

```diff
@@ -0,0 +1,250 @@
+        expectedEnd_1 = Optional.empty()
+        expectedEnd_2 = Optional.empty()
+        expectedEnd_3 = Optional.empty()
+
+        availability1.getExpectedStartDate(_ as PhysicalDataSourceConstraint) >> { expectedStart_1 }
+        availability1.getExpectedEndDate(_ as PhysicalDataSourceConstraint) >> { expectedEnd_1 }
+
+        availability2.getExpectedStartDate(_ as PhysicalDataSourceConstraint) >> { expectedStart_2 }
+        availability2.getExpectedEndDate(_ as PhysicalDataSourceConstraint) >> { expectedEnd_2 }
+
+        availability3.getExpectedStartDate(_ as PhysicalDataSourceConstraint) >> { expectedStart_3 }
+        availability3.getExpectedEndDate(_ as PhysicalDataSourceConstraint) >> { expectedEnd_3 }
+
+        availability1.getExpectedStartDate(_ as DataSourceConstraint) >> { expectedStart_1 }
+        availability1.getExpectedEndDate(_ as DataSourceConstraint) >> { expectedEnd_1 }
+
+        availability2.getExpectedStartDate(_ as DataSourceConstraint) >> { expectedStart_2 }
+        availability2.getExpectedEndDate(_ as DataSourceConstraint) >> { expectedEnd_2 }
+
+        availability3.getExpectedStartDate(_ as DataSourceConstraint) >> { expectedStart_3 }
+        availability3.getExpectedEndDate(_ as DataSourceConstraint) >> { expectedEnd_3 }
+
+        availability1.getDataSourceNames() >> ([DataSourceName.of("dsName_1")] as Set<DataSourceName>)
+        availability2.getDataSourceNames() >> ([DataSourceName.of("dsName_2")] as Set<DataSourceName>)
+        availability3.getDataSourceNames() >> ([DataSourceName.of("dsName_3")] as Set<DataSourceName>)
+
+        compositeAvailability = new SimpleCompositeAvailability(
+                Stream.of(
+                        availability1,
+                        availability2,
+                        availability3
+                )
+        )
+    }
+
+    @Unroll
+    def "If #desc availabilities have no start date, the composite has no start date"() {
+        given:
+        expectedStart_1 = Optional.ofNullable((DateTime) testExpectedStart_1)
+        expectedStart_2 = Optional.ofNullable((DateTime) testExpectedStart_2)
+        expectedStart_3 = Optional.ofNullable((DateTime) testExpectedStart_3)
+
+        Optional<DateTime> result_1 = compositeAvailability.getExpectedStartDate(Mock(PhysicalDataSourceConstraint))
+        Optional<DateTime> result_2 = compositeAvailability.getExpectedStartDate(Mock(DataSourceConstraint))
+
+        expect:
+        !result_1.isPresent()
+        !result_2.isPresent()
+
+        where:
+        testExpectedStart_1 |   testExpectedStart_2 |   testExpectedStart_3 |   desc
+        earliestStart       |   middleStart         |   null                |   "one"
+        earliestStart       |   null                |   null                |   "two"
+        null                |   null                |   null                |   "all"
+    }
+
+    @Unroll
+    def "If #desc availabilities have no end date, the composite has no end date"() {
```

Comment:
This test doesn't DO anything with it's parameters.

Reply:
A lot of the logic is happening in the setup function. `compositeAvailability` is being constructed out of 3 different Availability mocks. The parameters are setting the availability of those 3 mocks, and I am checking the behavior of the composite availability that wraps them.